### ISSUE TITLE
pytorch 1.0 speed is lower than 0.4?

### ISSUE THREAD

**Issue by Tshzzz** [pytorch/pytorch#15516]
the code running speed in pytorch0.4.1 is about 100 fps , but in pytorch 1.0 stable is only 50 fps.
I test on 1080Ti cuda9.2 and cudnn7.
`   

    import torch
    import torch.nn as nn
    import torch.nn.functional as F
    import os
    import torch.nn.init as init

    class L2Norm(nn.Module):
        def __init__(self, n_channels, scale):
            super(L2Norm, self).__init__()
            self.n_channels = n_channels
            self.gamma = scale or None
            self.eps = 1e-10
            self.weight = nn.Parameter(torch.Tensor(self.n_channels))
            self.reset_parameters()

        def reset_parameters(self):
            init.constant_(self.weight, self.gamma)

        def forward(self, x):
            norm = x.pow(2).sum(dim=1, keepdim=True).sqrt() + self.eps
            x = torch.div(x, norm)
            out = self.weight.unsqueeze(0).unsqueeze(2).unsqueeze(3).expand_as(x) * x
            return out
    class SSD(nn.Module):
        def __init__(self, size, base, extras, head, num_classes):
            super(SSD, self).__init__()
            self.num_classes = num_classes
            self.size = size
            self.vgg = nn.ModuleList(base)
            self.L2Norm = L2Norm(512, 20)
            self.extras = nn.ModuleList(extras)
            self.loc = nn.ModuleList(head[0])
            self.conf = nn.ModuleList(head[1])
    
        def forward(self, x):
    
            sources = list()
            loc = list()
            conf = list()
    
            # apply vgg up to conv4_3 relu
            for k in range(23):
                x = self.vgg[k](x)
    
            s = self.L2Norm(x)
            sources.append(s)
    
            for k in range(23, len(self.vgg)):
                x = self.vgg[k](x)
            sources.append(x)
    
    
            for k, v in enumerate(self.extras):
                x = F.relu(v(x), inplace=True)
                if k % 2 == 1:
                    sources.append(x)
    
            for (x, l, c) in zip(sources, self.loc, self.conf):
                loc.append(l(x).permute(0, 2, 3, 1).contiguous())
                conf.append(c(x).permute(0, 2, 3, 1).contiguous())
    
            loc = torch.cat([o.view(o.size(0), -1) for o in loc], 1)
            conf = torch.cat([o.view(o.size(0), -1) for o in conf], 1)
    
            loc = loc.view(loc.size(0), -1, 4)
    
            conf = conf.view(conf.size(0), -1, self.num_classes)
    
            pred = (conf,loc)
            return pred
    
        def load_weights(self, base_file):
            other, ext = os.path.splitext(base_file)
            if ext == '.pkl' or '.pth':
                print('Loading weights into state dict...')
    
                pred_dict = torch.load(base_file)
                model_dict = self.state_dict()
    
    
                for m in self.modules():
                    if isinstance(m, nn.Conv2d):
                        init.xavier_uniform_((m.weight.data))  # xavier_uniform_
                        m.bias.data.zero_()
    
                for (k, v), (k1, v1) in zip(pred_dict.items(), model_dict.items()):
                    if v.shape == v1.shape:
                        model_dict[k1] = v
    
    
                self.load_state_dict(model_dict)
    
                print('Finished!')
            else:
                print('Sorry only .pth and .pkl files supported.')
    
    
    def vgg(cfg, i, batch_norm=False):
        layers = []
        in_channels = i
        for v in cfg:
            if v == 'M':
                layers += [nn.MaxPool2d(kernel_size=2, stride=2)]
            elif v == 'C':
                layers += [nn.MaxPool2d(kernel_size=2, stride=2, ceil_mode=True)]
            else:
                conv2d = nn.Conv2d(in_channels, v, kernel_size=3, padding=1)
                if batch_norm:
                    layers += [conv2d, nn.BatchNorm2d(v), nn.ReLU(inplace=True)]
                else:
                    layers += [conv2d, nn.ReLU(inplace=True)]
                in_channels = v
        pool5 = nn.MaxPool2d(kernel_size=3, stride=1, padding=1)
        conv6 = nn.Conv2d(512, 1024, kernel_size=3, padding=6, dilation=6)
        conv7 = nn.Conv2d(1024, 1024, kernel_size=1)
        layers += [pool5, conv6,
                   nn.ReLU(inplace=True), conv7, nn.ReLU(inplace=True)]
        return layers
    
    
    def add_extras(cfg, i, batch_norm=False):
        # Extra layers added to VGG for feature scaling
        layers = []
        in_channels = i
        flag = False
        for k, v in enumerate(cfg):
            if in_channels != 'S':
                if v == 'S':
                    layers += [nn.Conv2d(in_channels, cfg[k + 1],
                                         kernel_size=(1, 3)[flag], stride=2, padding=1)]
                else:
                    layers += [nn.Conv2d(in_channels, v, kernel_size=(1, 3)[flag])]
                flag = not flag
            in_channels = v
        return layers
    
    
    def multibox(vgg, extra_layers, cfg, num_classes):
        loc_layers = []
        conf_layers = []
        vgg_source = [21, -2]
        for k, v in enumerate(vgg_source):
            loc_layers += [nn.Conv2d(vgg[v].out_channels,
                                     cfg[k] * 4, kernel_size=3, padding=1)]
            conf_layers += [nn.Conv2d(vgg[v].out_channels,
                                      cfg[k] * num_classes, kernel_size=3, padding=1)]
        for k, v in enumerate(extra_layers[1::2], 2):
            loc_layers += [nn.Conv2d(v.out_channels, cfg[k]
                                     * 4, kernel_size=3, padding=1)]
            conf_layers += [nn.Conv2d(v.out_channels, cfg[k]
                                      * num_classes, kernel_size=3, padding=1)]
        return vgg, extra_layers, (loc_layers, conf_layers)
    
    
    base = {
        '300': [64, 64, 'M', 128, 128, 'M', 256, 256, 256, 'C', 512, 512, 512, 'M',
                512, 512, 512],
        '512': [],
    }
    extras = {
        '300': [256, 'S', 512, 128, 'S', 256, 128, 256, 128, 256],
        '512': [],
    }
    mbox = {
        '300': [4, 6, 6, 6, 4, 4],  # number of boxes per feature map location
        '512': [],
    }
    
    
    def build_ssd300(size=300,pretrained = None ,num_classes=21):
        if size != 300:
            print("ERROR: You specified size " + repr(size) + ". However, " +
                  "currently only SSD300 (size=300) is supported!")
            return
        base_, extras_, head_ = multibox(vgg(base[str(size)], 3),
                                         add_extras(extras[str(size)], 1024),
                                         mbox[str(size)], num_classes)
    
        net = SSD(size, base_, extras_, head_, num_classes)
        if pretrained is not None:
            net.load_weights(pretrained)
    
        return net
    
    if __name__ == '__main__':
        import tqdm
        net = build_ssd300().cuda()
    
    
        for i in tqdm.tqdm(range(3000)):
    
            input = torch.randn(1, 3, 300, 300).cuda()
            loc, cls = net(input)

   `

**Comment by mosjoker** [comment]
Add the line in your code(PyTorch 1.0)
cudnn.benchmark = True

**Comment by Tshzzz** [comment]
> Add the line in your code(PyTorch 1.0)
> cudnn.benchmark = True

Thanks!
